### PR TITLE
Hapi patch

### PR DESCRIPTION
**Pull request by mcarans**
Add HAPI patch utility with test

### CODE REVIEW

**Review by turnerm — APPROVED** [review]
Thanks so much for coding this up Mike, looks good as far as I can tell, I may have more comments once I start to use it.

And don't forget the changelog :)

File: src/hapi/pipelines/utilities/hapi_patch.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        hapi_repo: str,
+        no_attempts: int = 100,
+        sleep_time: int = 60,
+        githubclass: Type[Any] = Github,
+    ) -> None:
+        self.hapi_repo = hapi_repo
+        self.no_attempts = no_attempts
+        self.sleep_time = sleep_time
+        self.lock_name = "LOCK"
+        self.lock_value = "LOCKED"
```
[the remark this answers is not in the document]
Comment (mcarans) [reply src/hapi/pipelines/utilities/hapi_patch.py:42]: Good idea!

File: tests/test_hapi_patch.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        MockRepo.variables = {}
+        MockRepo.files = [
+            MockFile("file", "hapi_patch_1_hno_afg.json"),
+            MockFile("dir", "test_dir"),
+            MockFile("file", "hapi_patch_2_hno_afg.json"),
+        ]
+        with HAPIPatch("test", githubclass=MockGithub) as hapi_patch:
+            val = hapi_patch.get_sequence_number()
+            assert val == 3
+        monkeypatch.delenv("GITHUB_TOKEN")
```
[the remark this answers is not in the document]
Comment (mcarans) [reply tests/test_hapi_patch.py:103]: Yes I've changed to use mocker

File: src/hapi/pipelines/utilities/hapi_patch.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+            patch (Any): Python patch object
+            **kwargs: Additional arguments to pass to PyGithub create_file
+
+        Returns:
+            None
+        """
+        filename = f"hapi_patch_{self.sequence_number}_{suffix}.json"
+        message = f"Creating {filename}"
+        logger.info(f"{message} in GitHub repository.")
+        content = json.dumps(patch, indent=None, separators=(",", ":"))
```
[the remark this answers is not in the document]
Comment (mcarans) [reply src/hapi/pipelines/utilities/hapi_patch.py:158]: The HNO files can be pretty big eg. 20 Mb, so will leave as None for now.

**Pull request merged**